common/djangoapps/embargo/tests/test_middleware.py

```diff
@@ -170,3 +170,21 @@ def test_whitelist_ip_skips_country_access_checks(self):
         # even though we would have been blocked by country
         # access rules.
         self.assertEqual(response.status_code, 200)
+
+    @patch.dict(settings.FEATURES, {'EMBARGO': True})
+    def test_always_allow_course_detail_access(self):
+        """ Access to the Course Structure API's course detail endpoint should always be granted. """
+        # Blacklist an IP address
+        ip_address = "192.168.10.20"
+        IPFilter.objects.create(
+            blacklist=ip_address,
+            enabled=True
+        )
+
+        url = reverse('course_structure_api:v0:detail', kwargs={'course_id': unicode(self.course.id)})
+        response = self.client.get(
```

Comment:
does this test need to use the `restrict_course` context manager, like the preceding one does?   https://github.com/edx/edx-platform/blob/clintonb/course-detail-whitelist/common/djangoapps/embargo/tests/test_middleware.py#L161